### PR TITLE
#564 - Added delegated grant protocol scope for RecordsRead, RecordsQuery, & RecordsDelete

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: tests/scenarios/delegated-grant.spec.ts [hunk 1]
```diff
@@ -235,21 +235,35 @@ export function testDelegatedGrantScenarios(): void {
...
       // Alice creates a delegated grant for device X to act as Alice.
-      const scope: PermissionScope = {
-        interface : DwnInterfaceName.Records,
-        method    : DwnMethodName.Write,
-        protocol
-      };
+      const queryGrantForDeviceX = await PermissionsGrant.create({
+        delegated   : true, // this is a delegated grant
+        dateExpires : Time.createOffsetTimestamp({ seconds: 100 }),
+        description : 'Allow device X to write as me in chat protocol',
```
Comment (LiranCohen) [inline tests/scenarios/delegated-grant.spec.ts:241]:
```suggestion
        description : 'Allow device X to query as me in chat protocol',
```

File: tests/scenarios/delegated-grant.spec.ts [hunk 2]
```diff
@@ -235,21 +235,35 @@ export function testDelegatedGrantScenarios(): void {
...
+        },
+        signer: Jws.createSigner(alice)
+      });
 
-      const grantToDeviceX = await PermissionsGrant.create({
+      // Alice creates a delegated grant for device X to act as Alice.
+      const readGrantForDeviceX = await PermissionsGrant.create({
         delegated   : true, // this is a delegated grant
         dateExpires : Time.createOffsetTimestamp({ seconds: 100 }),
         description : 'Allow device X to write as me in chat protocol',
```
Comment (LiranCohen) [inline tests/scenarios/delegated-grant.spec.ts:257]:
```suggestion
        description : 'Allow device X to read as me in chat protocol',
```

File: tests/scenarios/delegated-grant.spec.ts [hunk 3]
```diff
@@ -400,27 +414,25 @@ export function testDelegatedGrantScenarios(): void {
...
-        interface : DwnInterfaceName.Records,
-        method    : DwnMethodName.Write,
-        protocol
-      };
-
-      const grantToDeviceX = await PermissionsGrant.create({
+      const deleteGrantForDeviceX = await PermissionsGrant.create({
         delegated   : true, // this is a delegated grant
         dateExpires : Time.createOffsetTimestamp({ seconds: 100 }),
         description : 'Allow device X to write as me in chat protocol',
```
Comment (LiranCohen) [inline tests/scenarios/delegated-grant.spec.ts:420]:
```suggestion
        description : 'Allow device X to delete as me in chat protocol',
```

**Review by LiranCohen — COMMENTED** [review]
Looks good! Some small nits in the comments/grant labels.

File: tests/scenarios/delegated-grant.spec.ts (hunk 3 above)
Comment (thehenrytsai) [reply tests/scenarios/delegated-grant.spec.ts:420]: Thanks for the keen eyes! Fixed.

File: tests/scenarios/delegated-grant.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply tests/scenarios/delegated-grant.spec.ts:257]: Thanks for the keen eyes! Fixed.

**Review by LiranCohen — APPROVED** [review]
🚀 🚀

**Pull request merged**